core/schema/modulesource.go

```diff
@@ -269,17 +269,23 @@ func parseRefString(ctx context.Context, bk buildkitClient, refString string) pa
 func (s *moduleSchema) moduleSourceAsModule(
 	ctx context.Context,
 	src dagql.Instance[*core.ModuleSource],
-	args struct{},
+	args struct {
+		EngineVersion dagql.Optional[dagql.String]
+	},
 ) (inst dagql.Instance[*core.Module], err error) {
+	withSourceInputs := []dagql.NamedInput{
+		{Name: "source", Value: dagql.NewID[*core.ModuleSource](src.ID())},
+	}
+	if args.EngineVersion.Valid {
+		withSourceInputs = append(withSourceInputs, dagql.NamedInput{Name: "engineVersion", Value: args.EngineVersion})
+	}
```

Comment:
We *should* just be able to unconditionally attach this, but something goes bizarrely wrong if we do, and we just hang during a module call.

I spent ages trying to understand this. Apparently, if I attach the the arg *directly*, modules freeze up on initialization, but with no stack trace in an obviously guilty place and no errors anywhere in the logs. I checked with dagql, and there's nothing obviously incorrect about attaching an invalid optional like this.

No idea *why* or what could possibly be happening here... but this is a quick fix to unblock this. I wonder if maybe there's something potentially about maybe creating a module with a slightly different ID? cc @vito @sipsma maybe something stands out here?

Reply:
Yeah this might have sent you down a dusty corner of optional/`null` handling. I think this would result in an ID that has `"engineVersion": null` in it. Might be wise to have our ID construction just skip null optionals since it'll be considered different from the arg not being provided, for no good reason.

No idea what's up with the hang though, from a distance.